### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by krisselden**

### CODE REVIEW

File: lib/backburner/deferred-action-queues.ts [hunk 1]
```diff
@@ -3,27 +3,27 @@ import {
...
-    queues[queueName] = new Queue(queueName, options[queueName], options);
-  });
-}
+  private options: any;
 
-function noSuchQueue(name) {
-  throw new Error('You attempted to schedule an action in a queue (' + name + ') that doesn\'t exist');
-}
+  constructor(queueNames: string[], options: any) {
+    var queues = this.queues = {};
```
Comment (stefanpenner) [inline lib/backburner/deferred-action-queues.ts]: `let`

File: lib/backburner/queue.ts [hunk 2]
```diff
@@ -96,29 +47,9 @@ Queue.prototype = {
...
-        method.call(target);
-      }
-    } catch (error) {
-      onError(error, errorRecordedForStack);
-    }
-  },
+  }
 
-  flush: function(sync) {
+  public flush(sync?) {
```
Comment (stefanpenner) [inline lib/backburner/queue.ts]: `sync` should be an optional, but an optional boolean

**Comment by stefanpenner** [comment]
@krisselden should I review type related stuff, or is that for a later PR?

**Comment by krisselden** [comment]
I am doing it in pieces. Can we pick one or the other for the schedule API? Does anything else use?  Does ember data use later or setTimeout? Defer or schedule?

**Comment by stefanpenner** [comment]
@krisselden ember-data only uses `schedule` `join`:

```sh
 s/e/data ╍{}+ rg _backburner                                                                                                                                                                   (%1) (7s 786ms)
addon/-private/system/store.js
209:    this._backburner = new Backburner(['normalizeRelationships', 'syncRelationships', 'finished']);
2308:    let internalModelOrModels = this._backburner.join(() => {
2393:    this._backburner.schedule('normalizeRelationships', this, this._setupRelationships);
2665:    this._backburner.join(() => {
2666:      this._backburner.schedule('syncRelationships', this, this._flushUpdatedRelationships);
2758:    store._backburner.join(() => {

addon/-private/system/relationships/state/has-many.js
164:      this.store._backburner.join(() => {
```

My preference would be to reduce BB's API to only cater to ember, and deprecate the rest (and drop in 1.0.0). This goes especially for `setTimeout` which includes like 80loc to support both the `setTimeout` + `later` API's...

**Pull request merged**